query/PRoST-query/src/main/java/translator/Translator.java

```diff
@@ -361,21 +353,15 @@ private void removeTriplesFromGroups(final List<Triple> triples, final HashMap<S
 			entry.getValue().removeAll(triples);
```

Comment:
If subject of triples and object of a group (or the other way around) are not equal, there is no point to remove triples because they won't be any in common.

Reply:
497c18c12b302b2265c7ab5805307453490d8ebd